pystac/catalog.py

```diff
@@ -138,7 +139,7 @@ def save(self):
 
         STAC_IO.save_json(self.get_self_href(), self.to_dict())
 
-    def map_items(self, item_mapper):
+    def map_items(self, item_mapper, **kwargs):
```

Comment:
What's the inclusion of kwargs buy us here? Since we're passing in the method, the user can construct a method that closes over any parameters that don't involve the item. 

In your notebook example, I see:

```
spacenet_cog = spacenet.map_items(cogify_if_labeled, labels=labels, data_dir='/home/simon/Desktop/')
```

This could be written as:
```
spacenet_cog = spacenet.map_items(lambda item: cogify_if_labeled(item, labels=labels, data_dir='/home/simon/Desktop/'))
```
or
```
mod_cogs = lambda item: cogify_if_labeled(item, labels=labels, data_dir='/home/simon/Desktop/')
spacenet_cog = spacenet.map_items(mod_cogs)
```
Without the kwargs need (i.e we can take care of these needs with [partial application](https://en.wikipedia.org/wiki/Partial_application)).

Reply:
oh, just saw this. Yeah, that makes sense. I will change